### PR TITLE
Feature/hir 127 make company registration

### PR DESCRIPTION
**Pull request by AlexeyShok**
HIR-127. Second step for HIR-125, based on HIR-126. Make Company registration. Added endpoint to register a company. This endpoint update empty Company (we need to add it before adding Master Manager to maintain the integrity of the database).

### CODE REVIEW

**Review by kirillta — APPROVED** [review]

File: HappyTravel.Hiroshima.DirectManager/Services/ManagerManagementService.cs [hunk 1]
```diff
@@ -25,8 +26,10 @@ public Task<Result<Models.Responses.Manager>> Register(Models.Requests.Manager m
         {
            return Result.Success()
                 .Ensure(IdentityHashNotEmpty, "Failed to get the sub claim")
-                .Ensure(ContractManagerNotExist, "Contract manager has already been registered")
+                .Ensure(ManagerNotExist, "Contract manager has already been registered")
```
Comment (mingazhev) [inline HappyTravel.Hiroshima.DirectManager/Services/ManagerManagementService.cs:29]: Method returning bool should be named something like !DoesManagerExist

File: HappyTravel.Hiroshima.DirectManager/Services/ManagerManagementService.cs [hunk 2]
```diff
@@ -25,8 +26,10 @@ public Task<Result<Models.Responses.Manager>> Register(Models.Requests.Manager m
...
                 .Bind(() => IsRequestValid(managerRequest))
+                .Map(CreateCompany)
```
Comment (mingazhev) [inline HappyTravel.Hiroshima.DirectManager/Services/ManagerManagementService.cs:31]: Why this is splitted to two methods? Does it really matter in the whole logic chain to separate them?

File: HappyTravel.Hiroshima.DirectManager/Services/ManagerManagementService.cs [hunk 3]
```diff
@@ -61,20 +92,62 @@ Common.Models.Manager Create()
...
+                .Bind(company => Validate(companyRequest, company))
+                .Map(company => ModifyCompany(company))
+                .Map(Update)
+                .Map(Build);
+
+
+            Result<Common.Models.Company> Validate(Models.Requests.Company companyRequest, Common.Models.Company company)
+            {
+                var validationResult = ValidationHelper.Validate(companyRequest, new CompanyRegisterRequestValidator());
+                return validationResult.IsFailure ? Result.Failure<Common.Models.Company>(validationResult.Error) : Result.Success(company);
```
Comment (mingazhev) [inline HappyTravel.Hiroshima.DirectManager/Services/ManagerManagementService.cs:118]: Please place to separate lines

File: HappyTravel.Hiroshima.DirectManager/Services/ManagerManagementService.cs [hunk 4]
```diff
@@ -61,20 +92,62 @@ Common.Models.Manager Create()
...
 
-        
+
+        public Task<Result<Models.Responses.Company>> RegisterCompany(Models.Requests.Company companyRequest)
+        {
+            return _managerContext.GetManager()
+                .Ensure(manager => manager.IsMaster, "Manager has no rights to register the company")
+                .GetCompany(_dbContext)
+                .Bind(company => Validate(companyRequest, company))
+                .Map(company => ModifyCompany(company))
```
Comment (mingazhev) [inline HappyTravel.Hiroshima.DirectManager/Services/ManagerManagementService.cs:110]: Why these two methods are splitted?
From the overall logic looks like they should be as one step

File: HappyTravel.Hiroshima.DirectManager/Models/Requests/Company.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace HappyTravel.Hiroshima.DirectManager.Models.Requests
+{
+    public class Company
+    {
+        [Required]
```
Comment (mingazhev) [inline HappyTravel.Hiroshima.DirectManager/Models/Requests/Company.cs:12]: What does "Company" mean? Is it some kind of RegisterCompanyRequest?
And as I understand all the public request fields should be documented to be viewable through swagger

File: HappyTravel.Hiroshima.DirectManager/Services/ManagerManagementService.cs (hunk 2 above)
Comment (AlexeyShok) [reply HappyTravel.Hiroshima.DirectManager/Services/ManagerManagementService.cs:31]: Combining into one method violates the principle of single responsibility

File: HappyTravel.Hiroshima.DirectManager/Models/Requests/Company.cs (hunk 5 above)
Comment (AlexeyShok) [reply HappyTravel.Hiroshima.DirectManager/Models/Requests/Company.cs:12]: By convention, we do not use the word Request in the model for the request. We use the same names for the respective request, response and domain (database) models. Instead, it splits up requests, responses, and domain (database) models into three different subnamespaces in a common namespace.

**Pull request closed without merging**